### DISCUSSION TITLE
How to reveal objects hidden by other layers?

### DISCUSSION THREAD

**Discussion by user72356** [mapeditor/tiled#3305] [Ideas]
Maybe this could be an idea for an on/off toggle...

When scripting, is it possible to somehow link 2 object layers so that their content always moves together? I'm trying to create a system to show objects through the layers above them (to alert to hidden objects when editing). My idea was to duplicate the objects, place the duplicates/ghosts at the top of the layer pile, and set their opacity to 50%. But then obviously the system only works if the ghost objects always move together with the real objects. Thanks!

**Comment by bjorn** [comment]
Scripting could be a solution, for example by writing a manual action that duplicates your object layer to a read-only transparent mirror. The main issues I can think of is that currently there is no signal based on which you could automatically update this mirror, and also that changes to this mirror will be reflected in the undo stack and finally this mirror will get saved if you don't delete it first (though that could be automated using the [`assetAboutToBeSaved`](https://www.mapeditor.org/docs/scripting/modules/tiled.html#assetAboutToBeSaved) signal).

Apart from that, the built-in option is to reduce the opacity of the layers on top, either manually or by toggling "Highlight Current Layer" while having your object layer selected.

**Reply by user72356** [reply]
It was suggested to me that I instead make all non-object layers semi-transparent based on a scripted toggle, so that all objects are revealed. I haven't tried it but this may work.
